src/test/scala/com/cognite/sdk/scala/v1/FilesTest.scala

```diff
@@ -243,4 +256,30 @@ class FilesTest extends SdkTestSpec with ReadBehaviours with WritableBehaviors w
     assert(out.toByteArray().sameElements(expected))
     client.files.deleteById(file.id)
   }
+
+  it should "support search with dataSetIds" in {
+    val created = filesToCreate.map(f => client.files.createOneFromRead(f))
+    try {
+      val fromTime = created.map(_.createdTime).min
+      val toTime = created.map(_.createdTime).max
+      val foundItems = retryWhileEmpty {
+        client.files.search(FilesQuery(Some(FilesFilter(
+          dataSetIds = Some(Seq(CogniteInternalId(testDataSet.id))),
+          createdTime = Some(TimeRange(
+            min=fromTime,
+            max=toTime
+          ))
+        ))))
+      }
+      assert(!foundItems.isEmpty)
+      foundItems.foreach({ i =>
+        assert(i.dataSetId == Some(testDataSet.id))
+      })
+      created.filter(_.dataSetId.isDefined).foreach { c =>
```

Comment:
This too can be expressed more clearly using `forall` and/or one of the richer matchers in ScalaTest.

Reply:
1. I think that `forall` will yield an error message without any info in it, this should at least say *what* does not contain *what*
2. do you know which matcher? I could not find any "subset" matcher